modules/csg_2d/csg_shape_2d.cpp

```diff
@@ -0,0 +1,1046 @@
+
+				for (uint32_t i = 0; i < root_collision_shapes_convex.size(); i++) {
+					Ref<ConvexPolygonShape2D> convex_collision_shape;
+					convex_collision_shape.instantiate();
+					convex_collision_shape->set_points(root_collision_shapes_convex[i]->get_points());
+					baked_collision_shapes[i] = convex_collision_shape;
+				}
+			}
+
+			break;
+		case CSGShape2D::CollisionShapeType::COLLISION_SHAPE_TYPE_CONCAVE_SEGMENTS:
+			if (!root_collision_shapes_concave.is_empty()) {
+				baked_collision_shapes.resize(root_collision_shapes_concave.size());
+
+				for (uint32_t i = 0; i < root_collision_shapes_concave.size(); i++) {
+					Ref<ConcavePolygonShape2D> convex_collision_shape;
+					convex_collision_shape.instantiate();
+					convex_collision_shape->set_segments(root_collision_shapes_concave[i]->get_segments());
+					baked_collision_shapes[i] = convex_collision_shape;
+				}
+			}
+			break;
+		default: {
+			ERR_PRINT("CollisionShapeType failed. Unrecognized collision_shape_type.");
+			break;
+		}
+	}
+
+	return baked_collision_shapes;
+}
+
+Array CSGShape2D::bake_light_occluders() const {
+	Array baked_light_occluders;
+
+	if (!is_root_shape()) {
+		return baked_light_occluders;
+	}
+
+	for (uint32_t i = 0; i < mesh_outlines.size(); i++) {
+		if (mesh_outlines[i].size() < 3) {
+			continue;
+		}
+		Vector<Vector2> occluder_polygon = mesh_outlines[i];
+		occluder_polygon.push_back(occluder_polygon[0]);
+
+		Ref<OccluderPolygon2D> occluder_polygon_2d;
+		occluder_polygon_2d.instantiate();
+		occluder_polygon_2d->set_closed(false);
+		occluder_polygon_2d->set_polygon(occluder_polygon);
+
+		baked_light_occluders.push_back(occluder_polygon_2d);
+	}
+
+	return baked_light_occluders;
+}
+
+Ref<NavigationPolygon> CSGShape2D::bake_navigation_mesh() const {
+	Ref<NavigationPolygon> navigation_mesh;
+
+	if (!is_root_shape()) {
+		return navigation_mesh;
+	}
+
+	navigation_mesh.instantiate();
+
+	Vector<Vector<int>> nav_mesh_polygons;
+	nav_mesh_polygons.resize(mesh_convex_polygons.size());
+
+	for (uint32_t i = 0; i < mesh_convex_polygons.size(); i++) {
+		nav_mesh_polygons.write[i] = mesh_convex_polygons[i];
+	}
+
+	navigation_mesh->set_data(mesh_vertices, nav_mesh_polygons);
+
+	return navigation_mesh;
+}
+
+bool CSGShape2D::_is_debug_collision_shape_visible() {
+	return !Engine::get_singleton()->is_editor_hint() && is_inside_tree() && get_tree()->is_debugging_collisions_hint();
+}
+
+void CSGShape2D::_update_debug_collision_shape() {
+	if (!use_collision || !is_root_shape() || !root_collision_shapes_convex.is_empty() || !root_collision_shapes_concave.is_empty() || !_is_debug_collision_shape_visible()) {
+		return;
+	}
+
+	ERR_FAIL_NULL(RenderingServer::get_singleton());
+
+	if (root_collision_debug_instance.is_null()) {
+		root_collision_debug_instance = RS::get_singleton()->instance_create();
+	}
+
+	//RenderingServer::get_singleton()->canvas_item_add_mesh(get_canvas_item(),  root_edge_line_mesh->get_rid(), Transform2D(), color, texture_rid);
```

Comment:
I would add a `TODO` or a `FIXME` comment here, if it's not implemented in this PR.

Reply:
Those collision debug related functions are not really doing something. They exists to stay parallel with the CSG3D in case someone wants to do anything with it. I did not add special collision debug here because it would just create overlap noise in 2D as the default csg 2d visuals already show the edges compared to 3D.